sdk/python/tests/test_historical_retrieval.py

```diff
@@ -310,15 +314,20 @@ def test_historical_features_from_parquet_sources():
 @pytest.mark.parametrize(
     "provider_type", ["local", "gcp"],
 )
-def test_historical_features_from_bigquery_sources(provider_type):
+@pytest.mark.parametrize(
+    "infer_event_timestamp_col", [False, True],
```

Comment:
I like the e2e nature of this addition to the test, but as mentioned previously I think we should be cautious about further parameterization here, mostly because of execution times. That being said, I think this is the right approach for now. Ideally this kind of test would be incremental and wouldnt require the complete `test_historical_features_from_bigquery_sources` to rerun.

I will create an issue to track.